### PR TITLE
Allow to specify auditd rules in separate files

### LINKED ISSUE #7113: [Auditbeat] Allow audit rules to be specified in their own files
As a user I'd like to be able to specify my audit rules in dedicate file(s).

- Audit rules can be specified in their own file without having to worry about YAML string rules.
- Audit rules can be split across multiple files (e.g. `/etc/auditbeat/audit.rules.d/*.conf`) that are loaded in lexicographical order.

For confirmed bugs, please report:
- Version: Auditbeat 6.2
- Operating System: Linux
- Discuss Forum URL: https://discuss.elastic.co/t/auditbeat-conf-d-files/130661/5?u=andrewkroh

### PR DESCRIPTION
**Pull request by adriansr**
Added a new configuration field, `audit_rule_files` to the auditd
module. It allows to specify a set of files that contain audit rules.

```
- module: auditd
  audit_rule_files: [ '${path.config}/auditd.rules.d/*.conf' ]
```

Rules will be loaded into the kernel in the following order:
1. Rules specified in the `audit_rules` field at auditbeat.yml.
2. Rules from files in the `audit_rule_files`, in the order they are
   listed.
3. If a path contains a glob, the resulting files are loaded in
   lexicographical order.

Closes #7113

### CODE REVIEW

**Comment by exekias** [comment]
I recently unified how config reload works: https://github.com/elastic/beats/pull/7172

It would make sense to use or imitate this API, it would be especially useful to have support for file reloading, autodiscover or central management in the future. What do you think?

File: auditbeat/module/auditd/config.go [hunk 1]
```diff
@@ -39,17 +46,40 @@ type Config struct {
...
+	Warnings:               false,
+	ReassemblerMaxInFlight: 50,
+	ReassemblerTimeout:     2 * time.Second,
+	StreamBufferQueueSize:  8192,
+	StreamBufferConsumers:  0,
+}
+
 // Validate validates the rules specified in the config.
 func (c *Config) Validate() error {
+
```
Comment (andrewkroh) [inline auditbeat/module/auditd/config.go]: Shall we drop this line? :smile:

File: auditbeat/module/auditd/_meta/docs.asciidoc [hunk 2]
```diff
@@ -158,6 +161,10 @@ Defining any audit rules in the config causes {beatname_uc} to purge all
 existing audit rules prior to adding the rules specified in the config.
 Therefore it is unnecessary and unsupported to include a `-D` (delete all) rule.
 
+The rules declared in the `audit_rules` field are loaded first. Then the files
```
Comment (andrewkroh) [inline auditbeat/module/auditd/_meta/docs.asciidoc]: I think I would move this above with the reference information under `audit_rule_files`.

**Comment by andrewkroh** [comment]
> It would make sense to use or imitate this API, it would be especially useful to have support for file reloading, autodiscover or central management in the future.

@exekias I'm not 100% sure what you mean. Were you thinking that the auditd module should reuse the config reloading code? Or that it would be useful to have a generic file reloading capability (since the files here are not standard Beat YAML config files)?

I do think that we'll see requests for auto-reloading of the audit rules when they change.

**Comment by exekias** [comment]
Agreed, audit rules reloading is going to be really nice!

So https://github.com/elastic/beats/pull/7172 unifies how configs are reloaded for modules and inputs. The problem is split into 2: config files handling (detect changes, load them), and config reloading (take the new configs and do the proper changes to reach the desired state).

The second part may be useful here, and somehow prepares Auditbeat for the future of central management and autodiscover.

This is the current interface implementing config reloading: https://github.com/elastic/beats/pull/7172/files#diff-ab62770abd1b3493ace2c1a6c5426f0eR44 is used both by autodiscover & file config reload mechanisms as of today.

Happy to take this one into a call if needed. Also as I said, if you see no fit for these APIs I wouldn't block this PR, we can have a look to this in the future.

**Comment by adriansr** [comment]
@exekias @andrewkroh after looking at the required changes for config reloading I see that the required refactor is too big and I prefer not to block this PR, and have a separate PR for the config reloading.

**Comment by ruflin** [comment]
+1 on moving forward.

@andrewvc FYI: Seems we probably have one more implementation for config reloading from file.

**Comment by adriansr** [comment]
@andrewkroh I addressed your comment, please review again

**Pull request merged**